frontend/src/shared/helpers.tsx

```diff
@@ -804,9 +815,25 @@ export const CohortSampleDetailsColumns: ColDef[] = [
   },
 ];
 
-setupEditableSampleFields(SampleDetailsColumns);
+export const ReadOnlyCohortSampleDetailsColumns = _.cloneDeep(
+  CohortSampleDetailsColumns
+);
+
+setupEditableSampleFields(SampleMetadataDetailsColumns);
 setupEditableSampleFields(CohortSampleDetailsColumns);
 
+const seenColumns = new Set();
```

Comment:
this is kind of odd.  you should put this set up in some constructor.  also, you have a classic "side effect" here.  you are populating seenColumns inside of the filter function, which, per functional programming paradigm, should not mutate anything outside.  it should just return a value.    also, i think that the Set will take care of uniqueness by itself.  double check this but i think a Set will not allow duplicate values.

Reply:
@alisman Thanks, Aaron. Good call on the functional programming paradigm. 

Thoughts on the following code as replacement? All in one line so nothing outside is mutated. I confirmed that this achieves the same result.
```ts
export const combinedSampleDetailsColumns = 
    [...SampleMetadataDetailsColumns, ...ReadOnlyCohortSampleDetailsColumns]
        .filter((col, index, self) => index === 
            self.findIndex((c) => c.field === col.field));
```

Unfortunately, I couldn't use Set here because of 2 reasons:
1. The order of the columns in the data table is important
2. Set doesn't remove dups from an array of objects (even if the content of objects match completely) as each object has a unique reference

Reply:
i still find it very difficult to understand that code!  you're filtering for items (col) where the index is equal to the index of an item where field matches ...  ??   
you are just trying to remove duplicate objects which are not referentially equal?
if so, you could use _.uniqBy.    this would result in something much more readable.

Reply:
This is way cleaner and works. Awesome.

```ts
export const combinedSampleDetailsColumns = _.uniqBy(
  [...SampleMetadataDetailsColumns, ...ReadOnlyCohortSampleDetailsColumns],
  'field'
);
```